plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/RemoveUselessIsCheckFix.kt

```diff
@@ -6,28 +6,41 @@ import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import org.jetbrains.kotlin.diagnostics.Diagnostic
 import org.jetbrains.kotlin.idea.KotlinBundle
-import org.jetbrains.kotlin.psi.KtFile
-import org.jetbrains.kotlin.psi.KtIsExpression
-import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.idea.inspections.ConstantConditionIfInspection
+import org.jetbrains.kotlin.psi.*
 import org.jetbrains.kotlin.psi.psiUtil.getNonStrictParentOfType
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
 
-class RemoveUselessIsCheckFix(element: KtIsExpression) : KotlinQuickFixAction<KtIsExpression>(element) {
+class RemoveUselessIsCheckFix(
```

Comment:
It's absolutely strange to see that "constant `is` check" is implemented as a separate quick fix. We should have only one generic quick fix which is called smth like "SimplifyExpressionQuickFix". The quick fix should combinate `org.jetbrains.kotlin.idea.inspections.ConstantConditionIfInspection.SimplifyFix`, `org.jetbrains.kotlin.idea.inspections.ConstantConditionIfInspection.RemoveFix` and probably other quick fixes, IDK, it's up for investigation.

It's not your fault that we have a bit of dirty quick fixes design. And we have separate quick fix but since you are change behaviour in this subsystem it's better to refactor it before making any changes

Reply:
And it would be great to have dynamic quick fix message depending on the fix it's going to do. For example it should say "Unwrap 'if' statement" in case when it's going to unwrap if statement instead of generic "Simplify boolean expression"